### DISCUSSION TITLE
BSidebar Close on route change

### DISCUSSION THREAD

**Discussion by mngriffiths** [cdmoro/bootstrap-vue-3#2075] [Q&A] [answered]
The old Sidebar component defaulted to closing on route change https://bootstrap-vue.org/docs/components/sidebar#closing-on-route-change.
This component seems to have been replaced with Offcanvas, but I can't figure out how to duplicate the close on route change functionality

**Comment by Hiws** [comment, marked as answer]
You can use a watcher on the current route and hide your canvas when it changes.

```js
import { ref, watch } from "vue";
import { useRoute } from "vue-router";

const sidebarVisible = ref(false);

const route = useRoute();
watch(route, () => sidebarVisible.value = false);
```

**Reply by VividLemon** [reply]
> You can use a watcher on the current route and hide your canvas when it changes.

I don't really want to implement this on our end as it's separation of concerns issue. Use that example
